File: java/fibonacci.java
import java.util.Scanner;

public class fibonacci {
    public static void main(String[] args) {
        try (Scanner obj = new Scanner(System.in)) {
            int n = obj.nextInt();
            System.out.println(fibonacci_(n));
        }
    }

    public static long fibonacci_(int n) {
        return n > 0 ? fibonacci_(n - 1, 1, 1) : 0;
    }

    public static long fibonacci_(int n, long first, long second) {
        return n > 0 ? fibonacci_(n - 1, second, first + second) : first;
    }
}

File: c/fibonacci.c
#include <stdio.h>

unsigned long fibonacci(int n) {
    unsigned long first = 0, second = 1;
    for (; n > 0; n--) {
        unsigned long swap = first;
        first = second;
        second += swap;
    }
    return first;
}

int main() {
    int n;
    scanf("%d", &n);
    printf("%ld", fibonacci(n));
    return 0;
}
